sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala

```diff
@@ -187,10 +187,9 @@ object ResolveHints {
       def createRepartitionByExpression(
           numPartitions: Option[Int], partitionExprs: Seq[Any]): RepartitionByExpression = {
         val sortOrders = partitionExprs.filter(_.isInstanceOf[SortOrder])
-        if (sortOrders.nonEmpty) throw new IllegalArgumentException(
-          s"""Invalid partitionExprs specified: $sortOrders
-             |For range partitioning use REPARTITION_BY_RANGE instead.
-           """.stripMargin)
+        if (sortOrders.nonEmpty) {
+          throw QueryExecutionErrors.invalidRepartitionExpressionsError(sortOrders)
```

Comment:
it happens in analysis, should be query compilation error?

Reply:
`class IllegalArgumentException extends RuntimeException`
According to the description of https://issues.apache.org/jira/browse/SPARK-33539, RuntimeException => QueryExecutionErrors.
But the package is analysis.
cc @allisonwang-db What about your opinion ?

Reply:
ping @allisonwang-db

Reply:
Let's change it query compilation error first. It's definitely not a query execution error.

Reply:
This is a very good finding. I agree this should be a query compilation error, and the exception should be changed to AnalysisException eventually. We can probably also add a TODO here.

Reply:
OK